docs/import/color_function_util.js

```diff
@@ -266,49 +296,59 @@ function withColor(td, layer, property) {
   if (!colorFunction) {
     return td.text('N/A').addClass('na');
   }
-  const {currentStyle} = colorFunction;
-  switch (currentStyle) {
+  switch (colorFunction.currentStyle) {
     case ColorStyle.SINGLE:
-      td.append(createColorBox(colorFunction['color']));
+      td.append(createColorBox(colorFunction.color));
       break;
     case ColorStyle.CONTINUOUS:
-      td.append(createColorBox(colorFunction['color']));
+      td.append(createColorBox(colorFunction.color));
       break;
     case ColorStyle.DISCRETE:
       createColorBoxesForDiscrete(colorFunction, td);
       break;
     default:
       setStatus(ILLEGAL_STATE_ERR + 'unrecognized color function: ' + layer);
   }
-  td.addClass('editable color-td').on('click', () => onClick(td, currentStyle));
+  td.addClass('editable color-td').on('click', () => onClick(td));
   return td;
 }
 
 /**
  * On click function for color tds - updates the globalTd.
  * @param {JQuery<HTMLElement>} td
- * @param {enum} type
  */
-function onClick(td, type) {
+function onClick(td) {
   if ($(td).hasClass('na')) {
     return;
   }
+
   const colorFunctionDiv = $('#color-fxn-editor');
+  // open -> closed
   if (colorFunctionDiv.is(':visible') && td === globalTd) {
     colorFunctionDiv.hide();
     selectCurrentRow(false);
     return;
   }
-  colorFunctionDiv.show();
+
+  // most recent closed -> open
   if (td === globalTd) {
+    colorFunctionDiv.show();
     selectCurrentRow(true);
     return;
   }
+
+  // open td other than most recent closed
+  colorFunctionDiv.show();
   selectCurrentRow(false);
   globalTd = td;
   selectCurrentRow(true);
-  $('#' + colorStyleTypeStrings.get(type) + '-radio').prop('checked', true);
-  displaySchema(type);
+  const {currentStyle} = getColorFunction();
+  $('#' + colorStyleTypeStrings.get(currentStyle) + '-radio')
+      .prop('checked', true);
+  if (currentStyle !== ColorStyle.SINGLE) {
```

Comment:
why not unconditionally trigger changed for the radio above? Won't that do what you want here?

Reply:
not exactly, trigger the change for the continuous radio, for example, will do everything but check the by property radio. This is only an issue on color editor open since on regular schema change, you can't switch to continuous or discrete without the property radio already being checked.

Reply:
Ah, gotcha.